src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyNonCacheableResponse.java

```diff
@@ -0,0 +1,275 @@
+/**
+ * Copyright 2016-2019 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.http_cache.internal.stream;
+
+import org.agrona.DirectBuffer;
+import org.reaktivity.nukleus.function.MessageConsumer;
+import org.reaktivity.nukleus.http_cache.internal.proxy.request.Request;
+import org.reaktivity.nukleus.http_cache.internal.types.HttpHeaderFW;
+import org.reaktivity.nukleus.http_cache.internal.types.ListFW;
+import org.reaktivity.nukleus.http_cache.internal.types.OctetsFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.AbortFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.BeginFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.DataFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.EndFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.HttpBeginExFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.ResetFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.WindowFW;
+
+import static java.lang.System.currentTimeMillis;
+import static org.reaktivity.nukleus.http_cache.internal.HttpCacheConfiguration.DEBUG;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.HttpHeadersUtil.getHeader;
+
+final class HttpCacheProxyNonCacheableResponse extends HttpCacheProxyResponse
+{
+    private final HttpCacheProxyFactory streamFactory;
+
+    private MessageConsumer streamState;
+
+    private final MessageConsumer connectReplyThrottle;
+    private final long connectRouteId;
+    private final long connectReplyStreamId;
+
+    private Request streamCorrelation;
+    private long acceptInitialId;
+    private int acceptReplyBudget;
+
+    private long groupId;
+    private int padding;
+
+    private long traceId;
+
+    HttpCacheProxyNonCacheableResponse(
+        HttpCacheProxyFactory httpCacheProxyFactory,
+        MessageConsumer connectReplyThrottle,
+        long connectRouteId,
+        long connectReplyId,
+        long acceptInitialId)
+    {
+        this.streamFactory = httpCacheProxyFactory;
+        this.connectReplyThrottle = connectReplyThrottle;
+        this.connectRouteId = connectRouteId;
+        this.connectReplyStreamId = connectReplyId;
+        this.acceptInitialId = acceptInitialId;
+        this.streamState = this::beforeBegin;
```

Comment:
Remove `streamState` field, rename `handleStream` to `onResponseMessage`, add all cases to `onResponseMessage`, rename all case methods to `onBegin`, `onData`, `onEnd`, etc.